src/libsync/pushnotifications.h

```diff
@@ -129,7 +129,7 @@ private slots:
     void emitActivitiesChanged();
 
     Account *_account = nullptr;
-    QWebSocket *_webSocket = nullptr;
+    QWebSocket *_webSocket;
```

Comment:
If I understand, previously it could be there or missing and you are now always initializing it in the constructor. In this case, I would also use a plain object instead of a pointer. Especially because you are making the assumption in the code that it will be always valid.

Reply:
Yes, this would be better. However, there is one test I couldn't get working when this `_webSocket` it not a pointer. That is the test https://github.com/nextcloud/desktop/pull/3193#pullrequestreview-648296876. Any suggestions? (I know that is very ugly)

Reply:
No problem, just merge it now and we can do that later